pilot/pkg/xds/discovery.go

```diff
@@ -370,10 +370,9 @@ func debounce(ch chan *model.PushRequest, stopCh <-chan struct{}, opts debounceO
 	free := true
 	freeCh := make(chan struct{}, 1)
 
-	push := func(req *model.PushRequest, debouncedEvents int, startDebounce time.Time) {
+	push := func(req *model.PushRequest, debouncedEvents int) {
 		pushFn(req)
 		updateSent.Add(int64(debouncedEvents))
-		debounceTime.Record(time.Since(startDebounce).Seconds())
```

Comment:
This is changing the metric meaning

Reply:
yea, but the original metric is not measuring what it is supposed to measure. it is double counting the period between when we start to push until the push is actually finished. I draw this graph: currently it is measuring the blue time, while it should be measuring the green time
![Screenshot 2023-03-06 at 2 56 43 PM](https://user-images.githubusercontent.com/55459889/224119915-b7d25f85-6f45-4155-8ea5-d318f513ef34.png)

Reply:
According to the metric desc, it should be the blue time.

Reply:
I donot mind changing it, but i think we still need to monitor blue time - green time

Reply:
I see your point. I think the green time is valuable since it helps us measure the actual debounce time without the push (i.e. even though we set PILOT_DEBOUNCE_MAX 10s, the actual debounce time can take more than that)

So given that, maybe we can keep the blue one, and add another metric like debounce_stable_time?

Reply:
Don't we already have one for push time?

Reply:
https://github.com/istio/istio/blob/1.17.1/pilot/pkg/xds/monitoring.go#L118 this one is different, it's for XDS push, which is the time for building and sending the XDS resources.

Reply:
what do you guys think? if we want to keep the current metric unchanged to minimize impact to current users, how about adding another metric for recording the time to wait for events for each debouncing, something like DEBOUNCE_WAIT_TIME ?